packages/salesforcedx-vscode-soql/src/editor/queryRunner.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { JsonMap } from '@salesforce/ts-types';
+import { QueryResult } from 'jsforce';
+import * as vscode from 'vscode';
+
+export class QueryRunner {
+  constructor(
+    private connection: Connection,
+    private document?: vscode.TextDocument
+  ) {}
+
+  public async runQuery(queryText: string): Promise<JsonMap[]> {
+    try {
+      const rawQueryData = (await this.connection.query(
+        queryText
+      )) as QueryResult<JsonMap>;
+      const cleanQueryRecords = this.flattenQueryData(rawQueryData);
+
+      return cleanQueryRecords;
+    } catch (error) {
+      // TODO: Doc Review, i18n
+      vscode.window.showErrorMessage(`Could not run the Query ${error}`);
+      throw error;
+    }
+  }
+  /*
+  As query complexity grows
+  we will need to flatten the results of nested values
+  in order to be parsed and diplayed correctly
+  */
+  private flattenQueryData(rawqQueryData: QueryResult<JsonMap>) {
+    const records = rawqQueryData.records;
+    // filter out the attributes key
+    records.forEach(result => delete result.attributes);
+    return records;
+  }
+}
```

Comment:
The structure of the response from the API will contain attributes that we want to flatten in order to render the data correctly. For example, we currently support `Select field1, field2 FROM sObject` which yields the following data structure:
```
{
...,
  records: [
    {
      attributes: {
        type: 'sObject',
        url: ''
      },
      field1: '',
      field2: false
    }
  ]
}
```
we will need to remove the `attributes` property in order to render the table of data correctly. This is the same process that the CLI currently does, but with support for more nested data structures.

As the Query Builder supports more complex queries we will need to handle the response in a more robust way.